include/psp2/pss.h

```diff
@@ -8,29 +8,566 @@
 #define _PSP2_PSS_H_
 
 #include <psp2/types.h>
+#include <pthread.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct ScePssCryptoHandle {
-  SceUID fd;
-  uint32_t unk1;
-  SceSize size;
-  uint32_t unk3;
-} ScePssCryptoHandle;
-
-int pss_crypto_open(ScePssCryptoHandle *handle, char *path);
-char *pss_crypto_read(ScePssCryptoHandle *handle, int *mode);
-int pss_crypto_close(ScePssCryptoHandle *handle);
-void *pss_code_mem_alloc(SceSize *);
-void pss_code_mem_flush_icache(const void *, SceSize);
-void pss_code_mem_lock(void);
-void pss_code_mem_unlock(void);
+#define PSS_OK 0
+#define PSS_ERROR_THREAD_SUSPENDED -2147319763
 
-#ifdef __cplusplus
-}
-#endif
+/* IO Defines */
+#define PSS_ERROR_ERRNO_EEXIST -2147418095
+#define PSS_ERROR_ERRNO_ENOENT -2147418110
+#define PSS_ERROR_ERRNO_EACCES -2147418099
+
+#define PSS_STM_RWU (00600)
+#define PSS_STM_FDIR (0x1 << 12)
+#define PSS_STM_FREG (0x2 << 12)
+
+#define PSS_FREAD (0x0001)
+#define PSS_FWRITE (0x0002)
+#define PSS_FCREAT (0x0200)
+#define PSS_TRUNC (0x0400)
```

Comment:
maybe we should extract enums in `net.h`, `fcntl.h` and others to spererate files

at least, could you make the enum instead defines?

Reply:
im not sure what you mean ? move these defines to net.h?

Reply:
i mean we should split enums to a sperate file from `net.h` then share this values.